src/tree.h

```diff
@@ -17,15 +17,15 @@
 struct git_tree_entry {
 	uint16_t attr;
 	uint16_t filename_len;
-	git_oid oid;
-	bool pooled;
-	char filename[GIT_FLEX_ARRAY];
+	const git_oid *oid;
+	const char *filename;
 };
 
 struct git_tree {
 	git_object object;
+	git_odb_object *odb_obj;
+	git_array_t(git_tree_entry) entries_arr;
 	git_vector entries;
```

Comment:
I'm curious why we're keeping this vector here now that we have the array...  Is it just for keeping the entries sorted (differently than the array)?  And if so, is that necessary?